internal/node.bzl

```diff
@@ -53,12 +53,35 @@ def _write_loader_script(ctx):
           "TEMPLATED_module_roots": "\n  " + ",\n  ".join(module_mappings),
           "TEMPLATED_entry_point": ctx.attr.entry_point,
           "TEMPLATED_label_package": ctx.attr.node_modules.label.package,
-          # If the label being built is in another workspace, look for runfiles
-          # produced by that workspace
-          "TEMPLATED_workspace_name": (
+          # There are two workspaces in general:
+          # A) The user's workspace is the one where the bazel command is run
+          # B) The label's workspace contains the target being built/run
+          #
+          # If A has an npm dependency on B, then we'll look in the node_modules
+          # to find B's dependency D. It could be in two different places
+          # depending on hoisting:
```

Comment:
I tried to find a good reference for 'module hoisting' but all I got this sentence in https://docs.npmjs.com/cli/install

'If <folder> sits inside the root of your project, its dependencies may be hoisted to the toplevel node_modules as they would for other types of dependencies.'

Not quite the same use case. Are you sure 'hoisting' is the right term?

Reply:
Yeah, seems like hoisting is the term, might be yarn only. Will link to https://yarnpkg.com/lang/en/docs/workspaces/#toc-limitations-caveats